src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -56,6 +57,80 @@ pub struct TargetInfo {
     pub rustdocflags: Rc<[String]>,
 }
 
+/// Check Config (aka `--print=check-cfg` representation)
+#[derive(Debug, Default, Clone)]
+pub struct CheckCfg {
```

Comment:
Should this live in `cargo-platform`?

A valid answer can "maybe later" so we can keep it close for now as we figure out what this should look like

Reply:
Idk, but one thing to keep in mind is that `--print=check-cfg` is still unstable and I don't if we should expose unstable rustc parts in `cargo-platform`.

Reply:
While that flag is unstable, the formatting of check-cfg's is already stable and this is using the same format, right?

Reply:
Negative, the input of `--check-cfg` and the output of `--print=check-cfg` are different.
I choose to make the output of `--print=check-cfg` very close to `--print=cfg`, so it would be easier to parse for consumer.

Reply:
As suggested I put those in `cargo-platform` for now. [`f8179c5` (#14581)](https://github.com/rust-lang/cargo/pull/14581/commits/f8179c53c782dc10f6a6f77ec1a81d6acfb4bec8)

Reply:
> I choose to make the output of --print=check-cfg very close to --print=cfg, so it would be easier to parse for consumer.

Is there a place that documents differences?

I worry that having check-cfg and print-cfg be different will make things more difficult, not less, as we now have two distinct ways of talking about the same kind of thing.

Reply:
Yes, the documentation of [`--print=check-cfg`](https://doc.rust-lang.org/beta/unstable-book/compiler-flags/print-check-cfg.html), on the left is the `--check-cfg` input and on the right is the `--print=check-cfg` output.

My concern about re-using the `--check-cfg` syntax is that it's not easy to parse and would basically require every user to have a [MetaItem](https://doc.rust-lang.org/reference/attributes.html#meta-item-attribute-syntax) parser, while with the simplified output you only need to split the first `=` and strip the `"` for the value, much easier.

Reply:
I'm concerned we may add support for something that isn't expressible in this syntax, e.g. https://internals.rust-lang.org/t/pre-rfc-mutually-excusive-global-features/19618#additional-globalsvalues-validation-rules-24

Reply:
imo the name `--print=check-cfg` needs to change because that is not what its doing (but printing another format with the same intent) and giving it this name is confusing as I look through the PR.  I know in the context of this PR what it means **now** but in viewing the code as someone without that context, that isn't clear at all.

Similarly, this is terrible UI design for rustc's command-line.  The user has to not just read the docs but carefully parse a bullet list to try to understand that this is not check-cfg thats being printed.